src/core/connection.c

```diff
@@ -836,12 +837,37 @@ QuicConnGenerateNewSourceCid(
                 AllocFailure,
                 "Allocation of '%s' failed. (%llu bytes)",
                 "new Src CID",
-                sizeof(QUIC_CID_HASH_ENTRY) + MsQuicLib.CidTotalLength);
+                sizeof(QUIC_CID_SLIST_ENTRY) + MsQuicLib.CidTotalLength);
             QuicConnFatalError(Connection, QUIC_STATUS_INTERNAL_ERROR, NULL);
             return NULL;
         }
-        if (!QuicBindingAddSourceConnectionID(Connection->Paths[0].Binding, SourceCid)) {
-            CXPLAT_FREE(SourceCid, QUIC_POOL_CIDHASH);
+
+        BOOLEAN Collision = FALSE;
+
+        CxPlatDispatchLockAcquire(&MsQuicLib.DatapathLock);
+
+        // Check whether a sourceCid collides or not around all the connections.
```

Comment:
Can you clarify this comment. I don't quite understand. Also, can you update this to match the existing style for comments, so it has ``//` before and after too?

Reply:
//
        // Check whether a new sourceCid collides with any sourceCid which belong to all
        // the bindings including the ones which this connection doesn't bound.
        //

Does this make sense?